go/enclave/mempool/manager.go

```diff
@@ -93,10 +95,18 @@ func (db *mempoolManager) CurrentTxs(stateDB *state.StateDB) ([]*common.L2Tx, er
 			continue
 		}
 
-		if tx.Nonce() == nonceTracker.GetNonce(*sender) {
+		if tx.Nonce() != nonceTracker.GetNonce(*sender) {
+			continue
+		}
+
+		if err := limiter.AcceptTransaction(tx); err == nil { // Batch still has space
```

Comment:
this is quite a weird way to structure this I think, more standard to handle the two error cases first and then have the happy path not be inside an if block I think, like

```
if err != nil {
   if InsufficientSpace {
      ...
   }
   ...
}
// add transaction etc.
```